enhancements/api-review/custom-tags-aws.md

```diff
@@ -73,71 +95,376 @@ Note existing unchanged behavior: The installer will apply these tags to all AWS
+
+  The install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io` in `userTags`. In the existing design, kubernetes.io and openshift.io namespaces are not allowed as part of user-defined tags.
+
+  The `resourceTags` field in `.spec.platformSpec.aws` will be populated by the installer using the entries from `userTags` field.
+
+  `.spec.platformSpec.aws` is a mutable field and `.status.platformStatus.aws` is immutable.
+  `.status.platformStatus.aws` will have older version tags defined and is required for upgrade case.
+
+- All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api) will apply these tags to all AWS resources they create.
+  Operator must update AWS resource to match requested user-defined tags provided in `.spec.platformSpec.aws` (or `.status.platformStatus.aws`).
+
+  Operator must consider `.status.platformStatus.aws` to support upgrade scenarios.
+
+#### Create tags scenarios
+`resourceTags` that are specified in  `.spec.platformSpec.aws` of the Infrastructure resource will merge with user-defined tags in a kube resource.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already a tag with the same key present for AWS resource, the value from the AWS resource will be replaced.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value_comp1`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`
+
+   Action = Existing tag for AWS resource is updated to reflect new value.
+
+   Final tag set to AWS resource = `key_infra1 = value_infra1`
+
+   Event action = An event is generated to notify user about the request status (success/failure) to update tags for the AWS resource.
+
+2) There is no tag with same key present, new tag is created for the AWS resource. In case of limit reached, a validation error is generated.
+
+   For example,
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`
+
+   Action = A new tag is created for AWS resource.
+
+   Final tag set to AWS resource = `key_infra1 = value_infra1`
+
+   Event action = An event is generated to notify user about the request status (success/failure) to create tags for the AWS resource.
+
+#### Update tags scenarios
+Users can update the user-defined tags by editing `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type.
+On update of `resourceTags`, AWS resource is not created or restarted.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already a tag with the same key and value present for AWS resource, the tag for the AWS resource will not be updated.
+
+   For example,
+
+   Existing tag for AWS resource = `key_infra1 = value_update1`
+
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`
```

Comment:
these two values are the same.  should the first one be "value_compupdate1" and the second one "value_infraupdate1"?

Reply:
It is intentional that the two values are the same. This scenario is the following.
```
There is already a tag with the same key and value present for AWS resource
```
This scenario is demonstrating that adding the tag to the Infrastructure resource will not cause an update of the AWS resource when the AWS resource already has the matching tag.

We could argue whether this scenario is an interesting enough scenario to describe for a level-driven system.